singularity-eos/eos/eos_eospac.hpp

```diff
@@ -1274,43 +1274,52 @@ inline std::size_t EOSPAC::SetDynamicMemory(char *src, const SharedMemSettings &
 }
 inline std::size_t EOSPAC::SharedMemorySizeInBytes() const { return shared_size_; }
 
-SG_PIF_NOWARN
 template <typename Indexer_t>
 PORTABLE_INLINE_FUNCTION Real EOSPAC::TemperatureFromDensityInternalEnergy(
     const Real rho, const Real sie, Indexer_t &&lambda) const {
+#if defined(__CUDA_ARCH__) || __HIP_DEVICE_COMPILE__
```

Comment:
I'd prefer to put this logic in a single place and use it to set an ON_DEVICE macro. That way if the logic changes we have to change it in fewer places.

Reply:
I agree---do you know of a way to do that? can I define a "defined" macro as a preprocessor macro so I can do `#if PORTABLE_ON_DEVICE`?

Reply:
If you do something like this (probably better ways but this works):
```c++
#if defined(__CUDA_ARCH__) || defined(__HIP_ARCH__)
#define ON_DEVICE 1
#else
#define ON_DEVICE 0
#endif
...
#if ON_DEVICE
// do stuff
#else
// do host stuff
#endif 
```

Reply:
Hmm. I can try that. Let me try it first in just the eospac file since that's only where we need it and we can generalize and clean up later. Seem reasonable?

Also, apparently there's no `__HIP_ARCH__` its `__HIP_DEVICE_COMPILE__`

Reply:
Done.